src/read/gnu_compression.rs

```diff
@@ -0,0 +1,49 @@
+use crate::read::{self, Error, ReadError as _};
+use crate::{endian, CompressedFileRange, CompressionFormat, ReadRef as _, U32Bytes};
+
+// Attempt to parse the the CompressedFileRange for a section using the GNU-style
+// inline compression header format. This is used by the Go compiler in Mach-O files
+// as well as by the GNU linker in some ELF files. This function will return Ok(None)
+// if the section does not have the specified name prefix.
+pub(super) fn maybe_compressed_file_range<'data>(
```

Comment:
This will need `#[allow(unused)]` to avoid warnings with the elf and macho features disabled. Use `cargo xtask features` to check this.

Reply:
I instead marked this whole module as only existing when using macho or elf.